Fix tooltips in modals

Poltergeist opened the pull request:
This fixes the Tooltip overlay in modals by placing them on the left or the right of the icon instead of above. This was necessary because the container is now scrollable, thus is the overflow not visible.

If there are better Ideas how to solve this issue I am happy to hear them.

![image](https://cloud.githubusercontent.com/assets/156010/13643736/8488e004-e622-11e5-8735-c78761c37c3a.png)

![image](https://cloud.githubusercontent.com/assets/156010/13664864/dd82e31c-e6a9-11e5-9b01-a4f1b2d94b77.png)

![image](https://cloud.githubusercontent.com/assets/156010/13664873/e68363c4-e6a9-11e5-8639-39ad5eb7b21c.png)

![image](https://cloud.githubusercontent.com/assets/156010/13664852/c876df5a-e6a9-11e5-9de4-2d7c03f93b47.png)


![image](https://cloud.githubusercontent.com/assets/156010/13645113/9d836ae6-e629-11e5-8d89-8b496fa959e1.png)

mesosphere/marathon#3425

pierlo-upitup commented:
@Poltergeist thanks! A screenshot would be superhelpful, so that even @leemunroe can take a quick look!

Poltergeist commented:
Doh! you are right. Forgot to paste them earlier.

leemunroe commented:
Looks like you got it working having the tooltips above. Awesome! 

Maybe have @sascala do a quick grammar review. Noticing some inconsistencies.

sascala commented:
Happy to do a grammar check. But, @Poltergeist, the screenshots seem to show a previous version of the tooltips (from before my edits)?

Poltergeist commented:
I rebased this PR to the current master and updated the screenshots with the right tooltip messages. For ports and the JSON editor I couldn't find any tooltip text this is why I just placed something in there. For ports I did use the first part of the docs.

orlandohohmeier commented:
I'm impressed by your relentless search to somehow display the tooltips above the icons instead of displaying them "simply" on the left or the right of the icon. All the more I'm sorry that I have to point out that the current trick has some drawbacks as you can see in the following image:

![tooltip-position-issue](https://cloud.githubusercontent.com/assets/647035/13666106/2634c1a6-e6b0-11e5-9a66-e25b8e66762c.gif)

Let's revert to the  "simply" left/right tooltip version. 

/cc @leemunroe

Poltergeist commented:
:cry: :disappointed:

orlandohohmeier commented:
After reverting to the left/right tooltip we should also adjust the text alignment and maybe adjust the padding and arrow position. What do you think?

**Center**
![](https://cloud.githubusercontent.com/assets/647035/13645663/4a9a7cc2-e62c-11e5-9f04-b6be5c227539.png)

**Left**
![](https://cloud.githubusercontent.com/assets/647035/13645664/4a9e444c-e62c-11e5-9a95-89d67099c922.png)

/cc @leemunroe

pierlo-upitup commented:
The issue with the top-aligned tooltips was just the fixed-positioning? In this case are we sure that's not a fixable issue before reverting to left/right tooltips?

pierlo-upitup commented:
Yeah – "just" is an overly simplified way to put this. It's really not a trivial problem to solve.

leemunroe commented:
Damn. There's a discussion on it [here](https://github.com/Polymer/core-tooltip/issues/24) which I'm guessing you already came across.

Tooltip to the left/right sounds good. I would prefer to keep the style consistent though i.e. center aligned.

I think reducing the padding to `8px` (currently I think it is 16px) and upping border-radius to `3px` would look better.

orlandohohmeier commented:
@Poltergeist Have you tried using `position:fixed`? As we already use `getBoundingClientRect()` to determine whether is should be on top or beneath the icon, we could also use this to actually set the position relative to the viewport. 

Example: https://jsbin.com/cefiti/1/edit?output

aldipower commented:
Works like a charm and LGTM!
